src/sentinel.c

```diff
@@ -3302,12 +3304,36 @@ void sentinelConfigSetCommand(client *c) {
             sentinel.sentinel_auth_pass = sdslen(val->ptr) == 0 ?
                 NULL : sdsdup(val->ptr);
             drop_conns = 1;
+        } else if (!strcasecmp(option, "tls-cert-file") && moreargs > 0) {
+            val = c->argv[++i];
+            sdsfree(server.tls_ctx_config.cert_file);
+            server.tls_ctx_config.cert_file = sdslen(val->ptr) == 0 ?
+                                          NULL : sdsdup(val->ptr);
+            drop_conns = 1;
+        } else if (!strcasecmp(option, "tls-key-file") && moreargs > 0) {
+            val = c->argv[++i];
+            sdsfree(server.tls_ctx_config.key_file);
+            server.tls_ctx_config.key_file = sdslen(val->ptr) == 0 ?
+                                              NULL : sdsdup(val->ptr);
+            drop_conns = 1;
         } else {
             /* Should never reach here */
             serverAssert(0);
         }
     }
 
+    /* TLS certificate and key file were updated,  */
+    for (int i = 3; i < c->argc; i++) {
+        option = c->argv[i]->ptr;
+        if (!strcasecmp(option, "tls-key-file") || !strcasecmp(option, "tls-cert-file")) {
+            if (connTypeConfigure(connectionTypeTls(), &server.tls_ctx_config, 1) == C_ERR) {
+                addReplyErrorFormat(c, "Failed to configure TLS. Check logs for more info.");
+                goto exit;
+            }
+            break;
+        }
+    }
+
```

Comment:
No need to make the loop once again and lookup for tls related commands. Instead assist new flag `update_tls`